libs/full/version/CMakeLists.txt

```diff
@@ -10,20 +10,13 @@ set(version_headers hpx/version.hpp)
 
 set(version_sources version.cpp)
 
-if((HPX_WITH_NETWORKING AND HPX_WITH_PARCELPORT_MPI) OR HPX_WITH_ASYNC_MPI)
-  include(HPX_SetupMPI)
-  set(mpi_dependencies Mpi::mpi)
-endif()
-
 include(HPX_AddModule)
 add_hpx_module(
-  core version
+  full version
   GLOBAL_HEADER_GEN OFF
   SOURCES ${version_sources}
   HEADERS ${version_headers}
-  MODULE_DEPENDENCIES hpx_config hpx_config_registry hpx_format hpx_prefix
-  DEPENDENCIES Hwloc::hwloc ${mpi_dependencies}
-  MODULE_DEPENDENCIES hpx_config hpx_format hpx_prefix
+  MODULE_DEPENDENCIES hpx_config
```

Comment:
Isn't format still a dependency here?

Reply:
No. `hpx_format` is a module in `HPX::hpx_local`. The fact that it's a module in another library is something that `HPX::hpx` does not know about. Linking to it here would lead to duplicated symbols.